rotkehlchen/chain/ethereum/modules/uniswap/v3/utils.py

```diff
@@ -0,0 +1,463 @@
+import logging
+from typing import TYPE_CHECKING, Any, Dict, List, Set, Tuple
+
+from eth_abi import encode_abi
+from eth_abi.packed import encode_abi_packed
+from eth_utils import to_checksum_address
+from web3 import Web3
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.assets.utils import get_or_create_ethereum_token
+from rotkehlchen.chain.ethereum.contracts import EthereumContract
+from rotkehlchen.chain.ethereum.interfaces.ammswap.types import LiquidityPoolAsset
+from rotkehlchen.chain.ethereum.interfaces.ammswap.utils import TokenDetails
+from rotkehlchen.chain.ethereum.modules.uniswap.v3.types import NFTLiquidityPool
+from rotkehlchen.chain.ethereum.utils import multicall_2
+from rotkehlchen.constants.ethereum import (
+    UNISWAP_V3_FACTORY,
+    UNISWAP_V3_NFT_MANAGER,
+    UNISWAP_V3_POOL_ABI,
+)
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.misc import NotERC20Conformant, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEthAddress
+from rotkehlchen.utils.misc import get_chunks
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+UNISWAP_V3_POSITIONS_PER_CHUNK = 45
```

Comment:
How has the 45 number been chosen?

Reply:
With the accounts I tested, `45` was the number was not hit by `URL Too Long` error

Reply:
Still no answer here

Reply:
I tested with an account whereby the address has 81 LP positions.
> The amount should depend on number of assets? Or what?

It depends on the number of liquidity positions a user has(81 is quite an upper boundary as most users won't have up to that).